SupportYourLocals.WPF/MainWindow.xaml.cs

```diff
@@ -653,6 +658,265 @@ private void DisplayErrorMessage(string message)
+            {
+                LabelForEmptyFieldsError.Visibility = Visibility.Visible;
+                return false;
+            }
+            else if (csvData.GetData(UsernameTextBox.Text).Username != "")
+            {
+                LabelForUsedUsernameError.Visibility = Visibility.Visible;
+            }
+            else
+            {
+                if(!CheckUsernameRegex(username))
+                {
+                    LabelForInvalidUsernameError.Visibility = Visibility.Visible;
+                }
+                else // if username is correct
+                {
+                    if(!CheckPasswordLength(password))
+                    {
+                        LabelForPasswordError.Visibility = Visibility.Visible;
+                    }
+                    else if(CheckConfirmPassword(PasswordBoxR.Password, ConfirmPasswordBoxR.Password)) // if password is correct
+                    {
+                        return true;
+                    }
+                }
+            }
+            return false;
+        }
+
+        private bool CheckPasswordLength(String password)
+        {
+            if (password.Length >= 6)
+            {
+                return true;
+            }
+            return false;
+        }
+
+        private bool CheckUsernameRegex(String username)
+        {
+            Regex regexUsername = new Regex(@"^[a-zA-ZĄ-ž0-9-. ]+$");
+
+            if (!regexUsername.IsMatch(username))
+            {
+                return false;
+            }
+            return true;
+        }
+
+        private bool CheckConfirmPassword(string password, string confirmPassword)
+        {
+            if (password == confirmPassword)
+            {
+                return true;
+            }
+
+            LabelForConfirmPasswordError.Visibility = Visibility.Visible;
+            return false;
+        }
+
+        private void NewUser_Clicked(object sender, RoutedEventArgs e)
+        {
+            User.Visibility = Visibility.Visible;
+            LoginWhenRegistered.Visibility = Visibility.Collapsed;
+            GridLogin.Visibility = Visibility.Collapsed;
+            GridRegistration.Visibility = Visibility.Visible;
+        }
+
+        private void User_Clicked(object sender, RoutedEventArgs e)
+        {
+            GridLogin.Visibility = Visibility.Visible;
+            GridRegistration.Visibility = Visibility.Collapsed;
+        }
+
+        public bool CheckLoginData(String password, String username)
+        {
+            if (csvData.GetData(username).Username == username)
+            {
+                if (CompareHash(password, csvData.GetData(username).PasswordHash, csvData.GetData(username).Salt))
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        private bool CompareHash(string attemptedPassword, string base64Hash, string salt)
```

Comment:
This and other cryptography functions should either go to a separate class or become a part of the UserData class.